packages/analytics-browser/src/config.ts

```diff
@@ -168,85 +173,93 @@ export class BrowserConfig extends Config implements IBrowserConfig {
+  const lastEventId = previousCookies?.lastEventId;
+  const lastEventTime = previousCookies?.lastEventTime ?? legacyCookies.lastEventTime;
+  const lastSessionDeviceId = previousCookies?.deviceId ?? legacyCookies.deviceId;
+  const lastSessionUserId = previousCookies?.userId ?? legacyCookies.userId;
+  const optOut = options.optOut ?? previousCookies?.optOut ?? legacyCookies.optOut;
+  const sessionId = options.sessionId ?? previousCookies?.sessionId ?? legacyCookies.sessionId;
+  const userId = options.userId ?? previousCookies?.userId ?? legacyCookies.userId;
+
+  const trackingOptions = {
+    ...options.trackingOptions,
+    ipAddress: true,
+    language: true,
+    platform: true,
+  };
 
-  return new BrowserConfig(apiKey, {
-    ...options,
+  return new BrowserConfig(
+    apiKey,
+    options.appVersion,
     cookieStorage,
+    cookieOptions,
+    options.defaultTracking,
     deviceId,
-    domain,
+    options.flushIntervalMillis,
+    options.flushMaxRetries,
+    options.flushQueueSize,
+    identityStorage,
+    options.ingestionMetadata,
     lastEventId,
     lastEventTime,
+    lastSessionDeviceId,
+    lastSessionUserId,
+    options.loggerProvider,
+    options.logLevel,
+    options.minIdLength,
     optOut,
+    options.partnerId,
+    options.plan,
+    options.serverUrl,
+    options.serverZone,
     sessionId,
-    storageProvider: await createEventsStorage(options),
-    trackingOptions: {
-      ...defaultConfig.trackingOptions,
-      ...options?.trackingOptions,
-    },
-    transportProvider: options?.transportProvider ?? createTransport(options?.transport),
+    options.sessionTimeout,
+    options.storageProvider,
+    trackingOptions,
+    options.transport,
+    options.transportProvider,
+    options.useBatch,
     userId,
-  });
-};
-
-export const createCookieStorage = async <T>(
-  overrides?: BrowserOptions,
-  baseConfig = getDefaultConfig(),
-): Promise<Storage<T>> => {
-  const options = { ...baseConfig, ...overrides };
-  const cookieStorage = overrides?.cookieStorage as Storage<T>;
-  if (!cookieStorage || !(await cookieStorage.isEnabled())) {
-    return createFlexibleStorage<T>(options);
-  }
-  return cookieStorage;
+  );
 };
 
-const createFlexibleStorage = async <T>(options: BrowserOptions): Promise<Storage<T>> => {
-  let storage: Storage<T> = new CookieStorage({
-    domain: options.domain,
-    expirationDays: options.cookieExpiration,
-    sameSite: options.cookieSameSite,
-    secure: options.cookieSecure,
-  });
-  if (options.disableCookies || !(await storage.isEnabled())) {
-    storage = new LocalStorage();
-    if (!(await storage.isEnabled())) {
-      storage = new MemoryStorage();
-    }
+export const createCookieStorage = <T>(
+  identityStorage: IdentityStorageType = DEFAULT_IDENTITY_STORAGE,
+  cookieOptions: CookieOptions = {},
+) => {
+  if (identityStorage === DEFAULT_IDENTITY_STORAGE) {
```

Comment:
nit: I think it would be better to switch on the explicit value of identityStorage in case the default changes.

```suggestion
  switch (identityStorage) {
    case 'cookie':
      return new CookieStorage<T>(cookieOptions);
    case 'localStorage':
       ...
```

Reply:
Updated to switch statement with 'cookie' being default value